src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -351,18 +351,17 @@ ShouldHideShardsInternal(void)
 			return false;
 		}
 	}
-	else if (MyBackendType != B_BACKEND)
+	else if (MyBackendType != B_BACKEND && MyBackendType != B_WAL_SENDER)
```

Comment:
Yes, logical replication uses the replication connection (walsender process) to do queries on the catalogs. Those queries would return shards and that would cause failures on the subscriber side.